### ISSUE TITLE
Sponsorship Pages: Update URL structure

### ISSUE THREAD

**Issue by gracecanfield** [vigetlabs/npm#316]
## Overview

* **Summary:** Update the URL structure for Sponsorship pages so they are child pages of the Platform pages they relate to.  
* **Page or Element Name:** - Sponsorship pages (`/sponsorship/<slug>`)
* **Related Trello card:** https://trello.com/c/FGXLNijz/195-url-structure-for-sponsorship-pages-304
* **Estimate:** 4 hours (note: Please pause work and flag for Grace if you get into this and see that we will need more time.)
	
## Details

**Summary:** 

**Current vs. Desired state:**
Currently, sponsorship pages have the URL structure .../sponsorship/[page-title]. Desired state is .../[platform]/[page-title] - for example https://www.nationalpublicmedia.com/audio/radio-sponsorship/

**Current Sponsorship URLs**
* https://www.nationalpublicmedia.com/sponsorship/radio-sponsorship/
* https://www.nationalpublicmedia.com/sponsorship/podcast-sponsorship/
* https://www.nationalpublicmedia.com/sponsorship/voice-sponsorship/
* https://www.nationalpublicmedia.com/sponsorship/public-television/
* https://www.nationalpublicmedia.com/sponsorship/how-i-built-this-summit/
* https://www.nationalpublicmedia.com/sponsorship/tiny-desk-contest/

**Desired Sponsorship URLs:**
(Note: These are just examples. In reality, we'd want NPM should have control over which platform is used as the parent in the URL structure, for each sponsorship.)
* https://www.nationalpublicmedia.com/audio/radio-sponsorship/
* https://www.nationalpublicmedia.com/audio/podcast-sponsorship/
* https://www.nationalpublicmedia.com/audio/voice-sponsorship/
* https://www.nationalpublicmedia.com/video/public-television/
* https://www.nationalpublicmedia.com/events/how-i-built-this-summit/
* https://www.nationalpublicmedia.com/events/tiny-desk-contest/

**Approach:**

From @ten1seven originally, via Trello to the client:

>More realistically, we would recommend manually creating Pages that map to each Sponsorship entry/page, and we create a new "template" that duplicates the layout of the sponsorship detail page. The process would be:
>
>Create a page using a new "sponsorship" template at the desired URL.
>There would be only be a single dropdown on that admin page to select the sponsorship content to map to that URL.
>On the Viget side, we would simply duplicate the sponsorship template to use as a new page template.
>This has the benefit of working with how WordPress works. These pages would sit within the pages admin area for visibility and would be protected from accidental duplication by WordPress' built-in safeguards.

**Comment by gracecanfield** [comment]
@ten1seven Just a reminder to please keep this in a separate branch, silo'd from the other changes you worked on previously. Since this one will require client content entry, they don't want it to hold up getting the other changes out to production. Thanks!

**Comment by gracecanfield** [comment]
@ten1seven A couple other reminders:

* [ ] I checked with @aakashtandel and he said we shouldn't need to make any updates in GTM, but that you should just make sure the GTM.js script is added to the new URL page.
* [ ] Let's also try, if possible, to set up automatic redirects that go from the old to new URL structure. (Please let me know if we can't do this, so we know to add manually.)

**Comment by gracecanfield** [comment]
Got it; thanks! so to be clear, NPM &/or us will need to add any redirects manually in the CMS, right?

**Comment by gracecanfield** [comment]
Also, any sense of ETA on this? We'd originally told Whitney this would be ready for QA early this week

**Comment by ten1seven** [comment]
I just got the PR approved and will merge and deploy to staging for QA.

**Comment by ten1seven** [comment]
This is on staging for QA:

New page:
https://npm.staging.vigetx.com/audio/radio-sponsorship/

Admin page:
https://npm.staging.vigetx.com/wp/wp-admin/post.php?post=2598&action=edit

Changed nav menu entry from post type to page:
![Screen Shot 2020-04-07 at 11 12 37 AM](https://user-images.githubusercontent.com/137020/78699214-c5742a80-78c0-11ea-90d0-b2bad4601a34.png)

Added redirect:
![Screen Shot 2020-04-07 at 11 13 36 AM](https://user-images.githubusercontent.com/137020/78699270-df157200-78c0-11ea-9a64-7712b8fc7610.png)

**Comment by haleyjohnso** [comment]
To test this, I: 

- Navigated to the old URL /sponsorship/radio-sponsorship/ and it successfully redirected to /audio/radio-sponsorship/
- I opened up the production site /sponsorship/radio-sponsorship/ and compared it to staging /audio/radio-sponsorship/ 
- I did this comparison for a variety of browsers (firefox, Edge, Safari, IE11, Chrome) and devices (android, iphone, ipad)
- I verified that the changes also persist on the admin side

So far so good.

Then:

I notice that sponsorship/radio-sponsorship/ --> audio/radio-sponsorship is the only page that is set up with the sponsorship template (and redirect), but that the others exists and I can replace them on the page dropdown in the admin https://npm.staging.vigetx.com/wp/wp-admin/post.php?post=2598&action=edit

So I try to repeat the admin work for the podcast-sponsorship page. Everything seems to work:
1. I created a new page
2. made it child of Audio and added the podcast sponsorship template (saved it)
3. added the page to the menu
4. set up the the redirect for sponsorship/podcast-sponsorship/  (this is the only step I don't think i set up correctly because https://npm.staging.vigetx.com/audio/podcast-sponsorship doesn't redirect)

![Screen Shot 2020-04-07 at 10 20 46 PM](https://user-images.githubusercontent.com/57419582/78744194-2b42cf80-791e-11ea-854e-d48e76e14006.png)
 But https://npm.staging.vigetx.com/podcast-sponsorship does work even though https://npm.staging.vigetx.com/audio/podcast-sponsorship does not. 

Otherwise, it appears as though I've changed the URL structure for this page. 

Am I supposed to be replicating the admin work for the audio/radio-sponsorship for the other sponsorship templates?

**Comment by ten1seven** [comment]
@haleyjohnso I don't think you need to fully replicate all new pages and links as long as you think that what you've done is thorough enough.

Can you be more specific about this:

> But https://npm.staging.vigetx.com/podcast-sponsorship does work even though https://npm.staging.vigetx.com/audio/podcast-sponsorship does not.

If I click directly on those links I get to the same `/audio/podcast-sponsorship` page. Are you referring to links within the site that aren't redirecting correctly?

**Comment by haleyjohnso** [comment]
Wow, I think I was copying and pasting the wrong link. I meant https://npm.staging.vigetx.com/sponsorship/podcast-sponsorship wasn't working...but not it appears that it is

**Comment by ten1seven** [comment]
Try this in an incognito window ( or just not logged in to the admin) and it works. I don't know why some redirects work and this particular link doesn't when logged in, but it seems to work fine for non-admins.

**Comment by haleyjohnso** [comment]
Aha! Works in an incognito window

**Comment by haleyjohnso** [comment]
Also, just in case this isn't obvious--if we or NPM don't attach the templates to the right pages once we move to production, a lot of the pages 404. 

For example, I can't get to the Tiny Desk Concert sponsorship page on staging. https://npm.staging.vigetx.com/tiny-desk-contest/

Or by going to Platforms > Events > Tiny Desk Concert

**Comment by gracecanfield** [comment]
@ten1seven Hi there! It looks like this is ready to share with NPM. My main question before I do, is: What does the content entry approach look like for this? e.g., is there a way that we could get the back-end work on production so they can enter the content and then we go live with the front-end stuff? I'm trying to figure out a scenario where they don't have to be stuck with nonexistant or broken pages until they can enter the content.

**Comment by gracecanfield** [comment]
@ten1seven We were originally supposed to have this ready for NPM earlier this week, so I'm hoping to get the direction on content entry approach this morning so I can convey that along with staging review instructions. Any way you could follow up on [my question here](https://github.com/vigetlabs/npm/issues/316#issuecomment-611710071) when you get in today? Thanks!

**Comment by ten1seven** [comment]
I would need to backtrack on some work to allow both pages to exist. It would take ~30 minutes and I can do it this morning.

**Comment by ten1seven** [comment]
Okay, this is up on staging so you can verify it's working. You can now go to:

**Comment by gracecanfield** [comment]
@ten1seven Thank you so much! So is this saying that basically, we could deploy these to production and the user wouldn't see anything amiss? (the front-end would still be referencing the old templates?) And then would they have to enter the content, give us an okay, and then we would update the front-end to reference the new Page templates? 

Or if not, what would the steps be?

**Comment by ten1seven** [comment]
Correct on the first part. When they start adding the new pages and redirects it will all start working for visitors immediately. The only work I need to do to backfill is to _remove_ the ability for there to be pages for the sponsorship post type. If I left that in place, there would be duplicate pages (though "hidden" behind a redirect).

One nice side effect of making the sponsorship post types not able to generate their own page is that the Yoast SEO box stops displaying on the sponsorship post type edit pages. They asked about where they would edit the Yoast data and this eliminates any confusion (currently there is a Yoast box because the plugin understands that a page is being generated that it needs to account for).
